### PR TITLE
Add VerifyENSName endpoint

### PR DESCRIPTION
**Pull request by pedropombeiro**
This PR adds the `VerifyENSName` endpoint, to be used in https://github.com/status-im/status-go/issues/1520 to add some guarantees on user name authenticity.

It imports a new library to create namehash and invoke the `Pubkey` contract method  https://github.com/wealdtech/go-ens.

### CODE REVIEW

**Comment by pedropombeiro** [comment]
The test that is failing doesn't seem to be related to the changes in this PR.

File: api/backend.go [hunk 1]
```diff
@@ -555,6 +558,53 @@ func (b *StatusBackend) GetNodesFromContract(rpcEndpoint string, contractAddress
...
+func (b *StatusBackend) VerifyENSName(ensName, publicKeyStr, rpcEndpoint, contractAddress string) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), contractQueryTimeout)
+	defer cancel()
+
+	pubKeyBytes, err := hex.DecodeString(publicKeyStr)
+	if err != nil {
+		return false, err
+	}
+
+	expectedPubKey, err := btcec.ParsePubKey(pubKeyBytes, btcec.S256())
```
Comment (adambabik) [inline api/backend.go]: The same can be done with `crypto.UnmarshalPubkey()` where `crypto` is `github.com/ethereum/go-ethereum/crypto`.

File: api/backend.go [hunk 2]
```diff
@@ -555,6 +558,53 @@ func (b *StatusBackend) GetNodesFromContract(rpcEndpoint string, contractAddress
...
+		return false, err
+	}
+	x, y, err := resolver.PubKey()
+	if err != nil {
+		b.log.Error("error while resolving public key from ENS name", "ensName", ensName, "error", err)
+		return false, err
+	}
+
+	// Assemble the bytes returned for the pubkey
+	pb := make([]byte, btcec.PubKeyBytesLenUncompressed)
```
Comment (adambabik) [inline api/backend.go]: I would try to stick to `github.com/ethereum/go-ethereum/crypto`. Which means doing something like this:
```go
// crypto/elliptic
// github.com/ethereum/go-ethereum/crypto
pubKeyBytes := elliptic.Marshal(crypto.S256(), new(big.Int).SetBytes(x), new(big.Int).SetBytes(y))
return bytes.Equal(expectedPubKeyBytes, pubKeyBytes)
```

**Comment by yenda** [comment]
Is it an just an endpoint or does it also verify on new messages?

**Comment by pedropombeiro** [comment]
> Is it an just an endpoint or does it also verify on new messages?

For now this is just an enpoint. However I think I'll need to move this code to status-protocol-go and do the whole implementation there. Neither status-go nor status-react should care about verifying the authenticity of the sender.

**Pull request closed without merging**